pkgs/by-name/qq/qqmusic/package.nix

```
@@ -0,0 +1,138 @@
+{
+  fetchurl,
+  stdenv,
+  autoPatchelfHook,
+  makeWrapper,
+  lib,
+  makeDesktopItem,
+  copyDesktopItems,
+  # QQ Music dependencies
+  alsa-lib,
+  at-spi2-atk,
+  at-spi2-core,
+  cairo,
+  cups,
+  dbus,
+  expat,
+  gdk-pixbuf,
+  glib,
+  gtk3,
+  libdbusmenu,
+  libglvnd,
+  libpulseaudio,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  pciutils,
+  udev,
+  xorg,
+}:
+################################################################################
+# Mostly based on qqmusic-bin package from AUR:
+# https://aur.archlinux.org/packages/qqmusic-bin
+################################################################################
+stdenv.mkDerivation (finalAttrs: {
+  pname = "qqmusic";
+  version = "1.1.7";
+  src = fetchurl {
+    url = "https://dldir1.qq.com/music/clntupate/linux/qqmusic_${finalAttrs.version}_amd64.deb";
+    hash = "sha256-NPJHH7VwTzdNY87jFh28GaPjT7kRMweGI/XTOBAzM5E=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  buildInputs = [
+    alsa-lib
+    at-spi2-atk
+    at-spi2-core
+    cairo
+    cups
+    dbus
+    expat
+    gdk-pixbuf
+    glib
+    gtk3
+    libdbusmenu
+    libglvnd
+    libpulseaudio
+    mesa
+    nspr
+    nss
+    pango
+    pciutils
+    udev
+    xorg.libX11
+    xorg.libxcb
+    xorg.libXcomposite
+    xorg.libXcursor
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXi
+    xorg.libXrandr
+    xorg.libXrender
+    xorg.libXScrnSaver
+    xorg.libXtst
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    ar x $src
+    tar xf data.tar.xz
+
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    cp -r opt/qqmusic $out/opt
+    cp -r usr/* $out/
+
+    rm -rf $out/opt/swiftshader
+    ln -sf ${libglvnd}/lib $out/opt/swiftshader
+
+    mkdir -p $out/bin
+    makeWrapper $out/opt/qqmusic $out/bin/qqmusic \
+      --argv0 "qqmusic" \
```

```suggestion
      --inherit-argv0 \
```
Unless I'm mistaken, that will have the same effect. But you can leave it as is too.